src/main/java/com/aws/greengrass/smbridge/clients/MQTTClient.java

```diff
@@ -81,79 +82,63 @@ public void deliveryComplete(IMqttDeliveryToken token) {
+                         IMqttClient mqttClient) {
         this.mqttClientInternal = mqttClient;
         this.dataStore = new MemoryPersistence();
         this.serverUri = Coerce.toString(topics.findOrDefault(DEFAULT_BROKER_URI,
                 KernelConfigResolver.CONFIGURATION_CONFIG_KEY, BROKER_URI_KEY));
-        this.clientId = Coerce.toString(topics.findOrDefault(SMBridge.SERVICE_NAME,
+        this.clientId = Coerce.toString(topics.findOrDefault(DEFAULT_CLIENT_ID,
                 KernelConfigResolver.CONFIGURATION_CONFIG_KEY, CLIENT_ID_KEY));
         this.mqttClientKeyStore = mqttClientKeyStore;
         this.mqttClientKeyStore.listenToUpdates(this::reset);
+        this.executorService = executorService;
     }
 
-    private void reset() {
+    void reset() {
         if (mqttClientInternal.isConnected()) {
             try {
                 mqttClientInternal.disconnect();
             } catch (MqttException e) {
-                LOGGER.atError().setCause(e).log("Failed to disconnect MQTT Client");
+                LOGGER.atError().setCause(e).log("Failed to disconnect MQTT client");
                 return;
             }
         }
 
         try {
-            readKeyStoreAndConnect();
-        } catch (KeyStoreException | MqttException e) {
-            LOGGER.atError().setCause(e).log("Failed to connect with updated KeyStore");
-            return;
+            connectAndSubscribe();
+        } catch (KeyStoreException e) {
+            throw new RuntimeException(e);
         }
-
-        resubscribe();
-    }
-
-    private void readKeyStoreAndConnect() throws KeyStoreException, MqttException {
-        //TODO: persistent session could be used
-        connOpts.setCleanSession(true);
-
-        if (serverUri.startsWith("ssl")) {
-            SSLSocketFactory ssf = mqttClientKeyStore.getSSLSocketFactory();
-            connOpts.setSocketFactory(ssf);
-        }
-
-        LOGGER.atInfo().kv("uri", serverUri).kv(CLIENT_ID_KEY, clientId).log("Connecting to broker");
-        mqttClientInternal.connect(connOpts);
     }
 
     /**
      * Start the {@link MQTTClient}.
      *
-     * @throws MQTTClientException if unable to connect to the broker
+     * @throws RuntimeException if the client cannot load the KeyStore used to connect to the broker
      */
-    public void start() throws MQTTClientException {
+    public void start() throws RuntimeException {
+        mqttClientInternal.setCallback(mqttCallback);
         try {
-            // TODO: need retry logic here if we want to remove dependency on broker
-            readKeyStoreAndConnect();
-        } catch (MqttException | KeyStoreException e) {
-            LOGGER.atError().kv("uri", serverUri).kv(CLIENT_ID_KEY, clientId).log("Unable to connect to broker");
-            throw new MQTTClientException("Unable to connect to MQTT broker", e);
+            connectAndSubscribe();
+        } catch (KeyStoreException e) {
```

Comment:
Let's not throw a RuntimeException here. It's too generic. Also logging is missing

Reply:
Added logging. Continued convo about exception in above comment.